Gradle requires Git to start the app

We're teaching a JHipster Workshop at The Progressive Web Experience today and told students they didn't need to have Git installed. Creating the app w/o Git installed works, but running the app with `gradlew` (on Windows) fails if Git isn't installed.

##### **Overview of the issue**

Gradle requires Git to run.

##### **Motivation for or Use Case**

Git shouldn't be a requirement to run JHipster.

##### **Reproduce the error**

Create a new app with Gradle and run `gradlew` to start it.

##### **JHipster Version(s)**

6.5.1

##### **JHipster configuration**

```json
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "org.jhipster.blog",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "6.5.1",
    "applicationType": "monolith",
    "baseName": "blog",
    "packageName": "org.jhipster.blog",
    "packageFolder": "org/jhipster/blog",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "postgresql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "gradle",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "ZTg0OTUxYmE2YjNiMTc5NmNhZjFmZDg2YzExN2M1YzZjODVmN2Y3NWI3MzFmMmVlNjUyNWVjZWRlNjAzMDg0MTRhYTEyZjVhNGVhYzVlY2RkYjE4MzMyMGJhZjllNzEyODgwMjM3MDM0NGUzNThjMzk0MzMwOTg2NGFlMmJhOTA=",
    "embeddableLaunchScript": false,
    "useSass": true,
    "clientPackageManager": "npm",
    "clientFramework": "angularX",
    "clientTheme": "none",
    "clientThemeVariant": "",
    "testFrameworks": [
      "protractor"
    ],
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [],
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en",
      "es"
    ],
    "blueprints": []
  }
}
```

##### **Browsers and Operating System**

* Windows 7 Professional
* Java 11

-   [x] Checking this box is mandatory (this is just to show you read everything)

I was able to reproduce this on Windows 10 in Parallels. There is a warning about Git not being installed when you generate the app.

```
Installing languages: en, es
WARNING! git is not found on your computer.

WARNING! Git repository could not be initialized, as Git is not installed on your system
```

And the install process finishes successfully. There is an error about rimraf.

```
> blog@0.0.1-SNAPSHOT cleanup C:\Users\mraible\blog
> rimraf build/resources/main/static/ build/resources/main/aot

'rimraf' is not recognized as an internal or external command,
operable program or batch file.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! blog@0.0.1-SNAPSHOT cleanup: `rimraf build/resources/main/static/ build/resources/main/aot`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the blog@0.0.1-SNAPSHOT cleanup script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\mraible\AppData\Roaming\npm-cache\_logs\2019-12-03T16_44_57_393Z-debug.log
INFO! Congratulations, JHipster execution is complete!
```

And then it fails when you run `gradlew`:

```
> Task :compileJava
Note: Hibernate JPA 2 Static-Metamodel Generator 5.3.13.Final

> Task :generateGitProperties FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':generateGitProperties'.
> No .git directory found!

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
Use '--warning-mode all' to show the individual deprecation warnings.
See https://docs.gradle.org/6.0.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 7m 11s
6 actionable tasks: 5 executed, 1 up-to-date
```

It seems like the `generateGitProperties` task should be skipped when Git isn't installed.

--- a/generators/server/templates/build.gradle.ejs
+++ b/generators/server/templates/build.gradle.ejs
@@ -232,6 +232,11 @@ liquibase {
 }
 <%_ } _%>
 
+gitProperties {
+    failOnNoGitDirectory = false
+    keys = ["git.branch", "git.commit.id.abbrev", "git.commit.id.describe"]
+}
+
 checkstyle {
     toolVersion '${checkstyle_version}'
     configFile file("checkstyle.xml")

--- a/generators/server/templates/gradle/profile_prod.gradle.ejs
+++ b/generators/server/templates/gradle/profile_prod.gradle.ejs
@@ -75,16 +75,6 @@ processResources {
     <%_ } _%>
 }
 
-generateGitProperties {
-    onlyIf {
-        !source.isEmpty()
-    }
-}
-
-gitProperties {
-    keys = ["git.branch", "git.commit.id.abbrev", "git.commit.id.describe"]
-}
-
 <%_ if (!skipClient) { _%>
 test.dependsOn webpack_test
 processResources.dependsOn webpack

